volatility/framework/plugins/mac/lsmod.py

```diff
@@ -38,12 +38,29 @@ def list_modules(cls, context: interfaces.context.ContextInterface, layer_name:
         """
         kernel = contexts.Module(context, darwin_symbols, layer_name, 0)
 
+        kernel_layer = context.layers[layer_name]
+
         kmod_ptr = kernel.object_from_symbol(symbol_name = "kmod")
 
-        # TODO - use smear-proof list walking API after dev release
         kmod = kmod_ptr.dereference().cast("kmod_info")
-        while kmod != 0:
+
+        yield kmod
```

Comment:
Hmmmm, feels like the first one could presumably be invalid too?  Might be worth a check and then bombing out with an empty list if it is?

Reply:
True, will fix this pending your reply to my first comment on dereference(). I would like feedback on that first to make sure we are on the same page and/or if I missed something.